enterprise/backend/test/metabase_enterprise/sso/integrations/saml_test.clj

```diff
@@ -408,6 +413,44 @@
             (finally
               (db/delete! User :%lower.email "newuser@metabase.com"))))))))
 
+(deftest login-update-account-test
+  (testing "A new 'Unknown' name account will be created for a SAML user with no configured first or last name"
+    (do-with-some-validators-disabled
+      (fn []
+        (with-saml-default-setup
+          (try
+            (is (not (db/exists? User :%lower.email "newuser@metabase.com")))
+            ;; login with a user with no givenname or surname attributes
+            (let [req-options (saml-post-request-options (new-user-no-names-saml-test-response)
+                                                         (saml/str->base64 default-redirect-uri))]
+              (is (successful-login? (client-full-response :post 302 "/auth/sso" req-options)))
+              (is (= [{:email        "newuser@metabase.com"
+                       :first_name   "Unknown"
+                       :is_qbnewb    true
+                       :is_superuser false
+                       :id           true
+                       :last_name    "Unknown"
+                       :date_joined  true
+                       :common_name  "Unknown Unknown"}]
+                     (->> (mt/boolean-ids-and-timestamps (db/select User :email "newuser@metabase.com"))
+                          (map #(dissoc % :last_login))))))
+            ;; login with the same user, but now givenname and surname attributes exist
+            (let [req-options (saml-post-request-options (new-user-saml-test-response)
+                                                         (saml/str->base64 default-redirect-uri))]
+              (is (successful-login? (client-full-response :post 302 "/auth/sso" req-options)))
+              (is (= [{:email        "newuser@metabase.com"
+                       :first_name   "New"
+                       :is_qbnewb    true
+                       :is_superuser false
+                       :id           true
+                       :last_name    "User"
+                       :date_joined  true
+                       :common_name  "New User"}]
+                     (->> (mt/boolean-ids-and-timestamps (db/select User :email "newuser@metabase.com"))
+                          (map #(dissoc % :last_login))))))
+            (finally
+              (db/delete! User :%lower.email "newuser@metabase.com"))))))))
+
```

Comment:
This test:

1. Makes a new user (which does NOT have `:first_name` `:last_name`)
2. Checks that this newly created user has "Unknown" first/last name
3. Updates the user via an SSO login (a different SAML .xml file is loaded, which is the same user, but has first/last name attributes this time).
4. Check that the first/last name have been changed to match those provided by the identity provider.